### PR TITLE
Permalink

### LINKED ISSUE #8: Share builds via a static link

### PR DESCRIPTION
**Pull request by synopia**
- added permalink
- is generated, whenever Build object changes (link is updated immediately)
- in App::initState URL is checked for a given code and eventually load the build
- code is generated using datastream-js and lz-string, npm install is required
- fixes #8

### CODE REVIEW

File: src/builder/Build.kt [hunk 1]
```diff
@@ -20,6 +20,29 @@ class Build() {
     val totalHp get() = lane.getTotalHp(currentLevel)
     val totalDps get() = lane.getTotalDps(currentLevel)
 
+    fun save(ds:DSFactory.DataStream) {
+        if( legion!=null ) {
+            val index = LegionData.legions.indexOf(legion!!)
```
Comment (attrib) [inline src/builder/Build.kt]: Not better to use the legion id here? I'm unsure about this.

File: src/builder/data/Unit.kt [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
-        var upgradedLevel: Int? = null
-        var soldLevel: Int? = null
+    var buildLevel: Int? = null
+    var upgradedLevel: Int? = null
+    var soldLevel: Int? = null
+    var position = Position(0, 0)
+
+    fun save(ds:DSFactory.DataStream) {
+        val index = LegionData.units.indexOf(def)
+        ds.writeInt16(index)
```
Comment (attrib) [inline src/builder/data/Unit.kt]: same here. adding new units from legion side will kill old permalinks

**Review by attrib — CHANGES_REQUESTED** [review]
There are merge conflicts which also needs to be handled. (fixed by now)

Just a question the external libs are required by kotlin somehow them self? Or does the `@JsModule` takes care of the require?

Build returns an error:
Unexpected token: name (DataStream) [./node_modules/datastream-js/es/index.js:10,0][static/js/main.018dbaf1.js:15943,6]

Any clue? Does build work locally?

File: src/builder/Build.kt (hunk 1 above)
Comment (synopia) [reply src/builder/Build.kt]: I am unsure too. Using the string id for legion (same thing with units) will increase the permalink size by huge factor (~15 chars vs 2 byte). I'll test the string version with a big build and we'll see.

**Review by attrib — APPROVED** [review]

**Pull request merged**